core/services/adapters/PdfAdapter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace EventEspresso\core\services\adapters;
+
+use Dompdf\Autoloader;
+use Dompdf\Dompdf;
+use Dompdf\Options;
+
+/**
+ * Class PdfAdapter
+ *
+ * @package EventEspresso\core\services\adapters
+ */
+class PdfAdapter
+{
+    /**
+     * @var Options|null
+     */
+    private $options;
+
+    /**
+     * @var string
+     */
+    private $content;
+
+
+    /**
+     * PdfAdapter constructor.
+     */
+    public function __construct()
+    {
+        $this->options = null;
+        $this->content = '';
+    }
+
+
+    /**
+     * @return Options|null
+     */
+    public function getOptions()
+    {
+        // only load dompdf if nobody else has yet...
+        if (! class_exists('Dompdf\Autoloader') and is_readable(EE_THIRD_PARTY . 'dompdf/src/Autoloader.php')) {
+            require_once(EE_THIRD_PARTY . 'dompdf/src/Autoloader.php');
+            Autoloader::register();
+        }
+        if (! class_exists('Dompdf\Options')) {
+            return null;
+        }
+        $options = new Options();
+        $options->set('isRemoteEnabled', true);
+        $options->set('isJavascriptEnabled', false);
+        if (defined('DOMPDF_FONT_DIR')) {
+            $options->setFontDir(DOMPDF_FONT_DIR);
+            $options->setFontCache(DOMPDF_FONT_DIR);
+        }
+        // Allow changing the paper size.
+        if (defined('DOMPDF_DEFAULT_PAPER_SIZE')) {
+            $options->set('defaultPaperSize', DOMPDF_DEFAULT_PAPER_SIZE);
+        }
+        return $options;
+    }
+
+
+    /**
+     * @param Options $options
+     * @return $this
+     */
+    public function setOptions(Options $options): self
+    {
+        $this->options = $options;
+        return $this;
+    }
```

Comment:
In PHP, **ALL** objects are passed by reference. 

If some client code retrieves the Options object from this adapter, sets some values on it, then passes it back to this method, `$this->options = $options;` will simply be overwriting the property with the _exact same_ object. It is completely unnecessary to call this setter after setting options.

What _would_ be more useful (but completely unnecessary given our current use) would be to combine the logic you have in the `getOptions()` method with this `setOptions()` method, but instead of passing an `Options` instance to it, you pass an array of key value pairs that would get looped over and added to the  `Options` object. 

Maybe something like:

```php
    /**
     * @param array $extra_options
     * @return PdfAdapter
     */
    public function initializeOptions(array $extra_options): PdfAdapter
    {
        $this->options = new Options();
        $this->options->set('isRemoteEnabled', true);
        $this->options->set('isJavascriptEnabled', false);
        if (defined('DOMPDF_FONT_DIR')) {
            $this->options->setFontDir(DOMPDF_FONT_DIR);
            $this->options->setFontCache(DOMPDF_FONT_DIR);
        }
        // Allow changing the paper size.
        if (defined('DOMPDF_DEFAULT_PAPER_SIZE')) {
            $this->options->set('defaultPaperSize', DOMPDF_DEFAULT_PAPER_SIZE);
        }
        // now apply user provided options
        foreach( $extra_options as $key => $value ) {
                $this->options->set( $key, $value );
        }
        return $this;
    }
```

then your client code might be something like:

```php
        $pdf_adapter = new PdfAdapter();
        $options = [
                'key' => 'value'
        ];
        $pdf_adapter
                ->initializeOptions($options)
                ->setContent($content)
                ->generate($invoice_name . ".pdf", (bool) $download);
```

or since we are not actually setting any additional options:

```php
        $pdf_adapter = new PdfAdapter();
        $pdf_adapter
                ->initializeOptions()
                ->setContent($content)
                ->generate($invoice_name . ".pdf", (bool) $download);
```

great decision to use fluent setters btw 👍🏻